cmd/harmony/flags.go

```diff
@@ -894,6 +899,25 @@ func applyLogFlags(cmd *cobra.Command, config *harmonyConfig) {
 	}
 }
 
+var (
+	sysNtpServerFlag = cli.StringFlag{
+		Name:     "sys.ntp",
+		Usage:    "the ntp server",
+		DefValue: defaultConfig.Sys.NtpServer,
+	}
+)
+
+func applySysFlags(cmd *cobra.Command, config *harmonyConfig) {
+	if cli.HasFlagsChanged(cmd, sysFlags) {
+		cfg := getDefaultSysConfigCopy()
+		config.Sys = cfg
+	}
+
+	if cli.IsFlagChanged(cmd, sysNtpServerFlag) {
+		config.Sys.NtpServer = cli.GetStringFlagValue(cmd, sysNtpServerFlag)
+	}
```

Comment:
config loaded by harmony.conf is also parsed here, so it is possible that sys flags are not changed and ntp field is empty. Probably add:

```
if config.Sys.NtpServer == "" {
    config.Sys.NtpServer = defaultConfig.Sys.NtpServer
}
```

Do you think so?

Reply:
removed